[16.0][FIX] mrp_lot_number_propagation Keyerror stock.production.lot

bosd opened the pull request:
Fixes:
```
Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1653, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1680, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 1884, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 734, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 46, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 468, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 453, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/addons/mrp/models/mrp_production.py", line 1885, in button_mark_done
    productions_not_to_backorder._post_inventory(cancel_backorder=True)
  File "/mnt/data/odoo-addons-dir/quality_control_mrp_oca/models/mrp_production.py", line 33, in _post_inventory
    res = super()._post_inventory(cancel_backorder=cancel_backorder)
  File "/mnt/data/odoo-addons-dir/mrp_lot_production_date/models/mrp_production.py", line 11, in _post_inventory
    res = super()._post_inventory(cancel_backorder=cancel_backorder)
  File "/mnt/data/odoo-addons-dir/mrp_lot_number_propagation/models/mrp_production.py", line 102, in _post_inventory
    self._create_and_assign_propagated_lot_number()
  File "/mnt/data/odoo-addons-dir/mrp_lot_number_propagation/models/mrp_production.py", line 114, in _create_and_assign_propagated_lot_number
    lot_model = self.env["stock.production.lot"]
  File "/opt/odoo/odoo/api.py", line 550, in __getitem__
    return self.registry[model_name](self, (), ())
  File "/opt/odoo/odoo/modules/registry.py", line 186, in __getitem__
    return self.models[model_name]
KeyError: 'stock.production.lot'

The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    RPC_ERROR
        at makeErrorFromResponse (http://oca-manufacture-16-0-3271ae22e786.runboat.odoo-community.org/web/assets/608-565ca3b/web.assets_backend.min.js:1001:163)
        at XMLHttpRequest.<anonymous> (http://oca-manufacture-16-0-3271ae22e786.runboat.odoo-community.org/web/assets/608-565ca3b/web.assets_backend.min.js:1009:13)

```

bosd committed:
[FIX] mrp_lot_number_propagation Keyerror stock.production.lot

sebalix reviewed (approved):
Ouch, thanks!

sebalix reviewed (commented):
Still, this issue wasn't raised by the CI because the test module `test_mrp_production.py` wasn't imported here: https://github.com/OCA/manufacture/blob/16.0/mrp_lot_number_propagation/tests/__init__.py

Can you fix that?

bosd committed:
[FIX] mrp_lot_number_propagation: tests

import test_mrp_production
stock.move.line : product_uom_qty -> reserved_uom_qty

bosd commented:
Done!

sebalix commented:
Thanks!

/ocabot merge patch
